packages/gatsby-remark-autolink-headers/src/index.js

```diff
@@ -20,13 +20,30 @@ module.exports = (
     className = `anchor`,
     maintainCase = false,
     removeAccents = false,
+    enableCustomId = false,
   }
 ) => {
   slugs.reset()
 
   visit(markdownAST, `heading`, node => {
-    const slug = slugs.slug(toString(node), maintainCase)
-    const id = removeAccents ? deburr(slug) : slug
+    let id
+    if (enableCustomId && node.children.length > 0) {
+      const last = node.children[node.children.length - 1]
+      const match = /^(.*?)\s*\{#([\w-]+)\}$/.exec(toString(last))
```

Comment:
This seems a little messy. Is this code from elsewhere that's been copy/pasted or rather, can this be optimized/tweaked?

Reply:
Do you mean the regex looks messy or the entire code structure is too complicated?
In the former case, I will write unit tests for the regex.

> Is this code from elsewhere that's been copy/pasted
No, I implemented this from scratch.

Reply:
Just the combination of the regex and the other code there.

Maybe we can implement something like in [reactjs.org](https://github.com/reactjs/reactjs.org/blob/master/plugins/gatsby-remark-header-custom-ids/index.js)

Reply:
I hesitated to look at another repository maintained under a different copyright and I had not looked into reactjs.org version before your comment.

Anyway, the major difference between my PR and the code you mentioned is that

- my PR extracts `#{id}` only from the last node.
- reactjs.org version extract `#{id}` from the entire header node text.

For this markdown example, which I added to the test cases:

```
# Invalid {#this_is_italic}
```

My version does NOT extract a custom ID because `_is_` is italic. reactjs version extracts `#thisisitalic` and does NOT remove `{#this_is_` part from the header text.
I think the behavior of my version is easy to understand for users.

What do you think?